tests/robot-tests/scripts/create_snapshots.py

```diff
@@ -219,18 +221,57 @@ def _create_methodologies_snapshot(self) -> str:
             themes.append(theme)
         return json.dumps(themes, sort_keys=True, indent=2)
 
-    def validate_snapshots(self):
+    def _build_slack_blocks(self, messages: List[str]) -> List | None:
+        # if length of messages is 0, then no errors were found
+        blocks = []
+        if len(messages) > 0:
+            # PR is raised due to snapshot mismatch
+            blocks.append(
+                {
+                    "type": "header",
+                    "text": {
+                        "type": "plain_text",
+                        "text": "Snapshots do not match . See pull request for more details",
+                    },
+                }
+            )
+
+            blocks.append(
+                {
+                    "type": "actions",
+                    "elements": [
+                        {
+                            "type": "button",
+                            "style": "danger",
+                            "text": {"type": "plain_text", "text": "View pull request"},
+                            "url": "https://github.com/dfe-analytical-services/explore-education-statistics/pulls/dfe-sdt",
+                        }
+                    ],
+                }
+            )
+        return blocks if len(blocks) > 0 else None
+
+    def validate_snapshots(self) -> None:
+        messages: List[str] = []
+
         for name in self.snapshots:
             if self._validate_snapshot(name):
                 print(f"Snapshot {name} matches for {self.public_url}")
             else:
                 print(f"Snapshot {name} does not match for {self.public_url}")
-                if self.slack_webhook_url is not None:
-                    self._send_slack_notification(f"Snapshot {name} does not match for {self.public_url}")
-        if self.slack_webhook_url is not None:
-            self._send_slack_notification(f"Snapshot test workflow successfully completed for {self.public_url} ✅")
+                messages.append(f"Snapshot `{name}` does not match for {self.public_url}")
+
+        if len(messages) > 0 and self.slack_webhook_url:
+            # At least one snapshot did not match so send a failure message to slack
+            self._send_slack_notification(blocks=self._build_slack_blocks(messages))
 
-    def write_snapshots_to_file(self):
+        else:
```

Comment:
This `else` is going to be executed if there are differences, but `self.slack_webhook_url` is `None`. So even though there are failures, it's  going to print "Snapshot script complete. No differences found."

I think I'd do this instead:
```
if len(messages) > 0:
  self._send_slack_notification(blocks=self._build_slack_blocks(messages)) if self.slack_webhook_url else print("waah! snapshots don't match!")
else:
  ...
```